chore: deprecate ELASTIC_OTEL_METRICS_DISABLED (#768)
elastic/elastic-otel-node: packages/opentelemetry-node/CHANGELOG.md

## Unreleased

> - feat: add env var `ELASTIC_OTEL_HOST_METRICS_DISABLED` to controll wether

Reviewer: ```suggestion
- feat: Add env var `ELASTIC_OTEL_HOST_METRICS_DISABLED` to control whether
```

>   EDOT has to send metrics from the host (process, system, ...) or not. This

Reviewer: ```suggestion
  EDOT Node.js will collect host metrics (`process.*`). This
```

We don't include `system.*` metrics, so shouldn't include that here.

  means users can disable host metrics without affecting metrics from instrumentations.
  (https://github.com/elastic/elastic-otel-node/issues/736)

>   The existing `ELASTIC_OTEL_METRICS_DISABLED` becomes deprecated since the
>   disabling of metrics can be done via `OTEL_METRICS_EXPORTER` configuration
>   option.

Reviewer: ```suggestion
- The `ELASTIC_OTEL_METRICS_DISABLED` environment variable is now deprecated.
  Use `OTEL_METRICS_EXPORTER=none` (instead of `ELASTIC_OTEL_METRICS_DISABLED=true`)
  to disable any metrics being exported by EDOT Node.js.
```
